Fixed the USB logger

rmeertens opened the pull request:
The USB logger works again! Made it into the current setup where you can specify which camera to log.

gautierhattenberger commented:
it looks fine but I can't really test it

flixr commented on sw/airborne/modules/computer_vision/video_usb_logger.c:
```diff
@@ -28,35 +28,118 @@
 #define VIDEO_USB_LOGGER_PATH /data/video/usb
 #endif
 
+#ifndef VIDEO_LOG_WIDTH
```
please document newly added define options in the module xml

flixr commented on conf/modules/video_usb_logger.xml:
```diff
@@ -7,12 +7,14 @@
+  <periodic fun="video_usb_logger_periodic()" start="video_usb_logger_start()" stop="video_usb_logger_stop()" autorun="TRUE"/>
   <makefile target="ap">
     <file name="video_usb_logger.c"/>
 
```
please remove unused defines....

flixr commented on sw/airborne/modules/computer_vision/video_usb_logger.c:
```diff
@@ -28,35 +28,118 @@
+    // Create a high quality image (99% JPEG encoded)
+    jpeg_encode_image(img, img_jpeg, 99, TRUE);
+
+#if JPEG_WITH_EXIF_HEADER
```
this is an option that you should define in your module/airframe file?
Maybe it should be specific to this module?

flixr commented on sw/airborne/modules/computer_vision/video_usb_logger.c:
```diff
@@ -28,35 +28,118 @@
+struct image_t *log_image(struct image_t *img)
+{
+  save_shot(img, &img_jpeg);
+  return FALSE;
```
should probably return `img`

rmeertens commented:
Added all documentation!

flixr commented:
Thx, but I still think it would be good to use a common prefix for the defines, right now there are
- VIDEO_USB_LOGGER_
- VIDEO_USBLOG_
- VIDEO_LOG_
- and no prefix for JPEG_WITH_EXIF_HEADER

flixr commented:
also why is it named udb_logger in the first place? It has nothing to do with USB...

rmeertens commented:
Changed the names of the defines.
Added Christophe as author before my changes (I think it is important to see who worked on it in case of questions)
For now, let's keep it at USB logger. Originally the path was set the the USB stick of the ARDrone.

flixr commented:
thx!
